configuration/pih/htmlforms/section-obgyn-initial.xml

```diff
@@ -544,13 +544,23 @@
 			});
 			
 			beforeSubmit.push(function() {
-			var numErrors = 0;
-			jq('.medicalHistoryText').find('input[type=text]').each(function() {
-				if (!validateFamHistoryText(jq(this)) ) {
-				++numErrors;
-			}
+				var numErrors = 0;
+				jq('.medicalHistoryText').find('input[type=text]').each(function() {
+					if (!validateFamHistoryText(jq(this)) ) {
+					++numErrors;
+					}
+				});
+				return ( numErrors == 0 );
+			});
+
+			jq('#intake-or-followup').change(function() {
+				setRequiredIfVisible('knowing_period_question');
+			});
+			jq('#knowing_period_question').change(function() {
+				setRequiredIfVisible('knowing_period_question');
```

Comment:
Why do we run this method when these two things elements change?

Reply:
Ah, I think i get it, in this case the method is just being used to display the "Required" text if needed? (hence setRequiredIfNeeded method name?)

Reply:
Yes, that is the main intent.  But it also returns a boolean indicating whether it is currently required and not completed, which allows the same method to be used in the `beforeSubmit` validations that prevent form submission